### PR TITLE
En_kusa OK and Documented

### PR DESCRIPTION
**Pull request by jpburnett**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 1]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+
+/**
+ * @brief Waits for object to be loaded then draws the appropriate actor
+ *
+ */
+void EnKusa_WaitObject(EnKusa* this, GlobalContext* globalCtx) {
+    s16 pad;
+
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objIndex) != 0) {
+        s32 kusa_type = GET_KUSA_TYPE(this);
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:442]: Style is camelCase, remember.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 2]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnKusa_SetupWaitObject(EnKusa* this) {
+    this->actionFunc = EnKusa_WaitObject;
+}
+
+/**
+ * @brief Waits for object to be loaded then draws the appropriate actor
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:435]: It's all the same actor, you probably mean "model" or similar.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 3]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+ *
+ */
+void EnKusa_WaitObject(EnKusa* this, GlobalContext* globalCtx) {
+    s16 pad;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:439]: Does it have to be an s16?

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 4]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+void EnKusa_SetupMain(EnKusa* this) {
+    this->actionFunc = EnKusa_Main;
+    this->actor.flags &= ~0x10;
+}
+
+/**
+ * @brief Handles the main logic and control flow for all kusa types
+ *
+ */
+void EnKusa_Main(EnKusa* this, GlobalContext* globalCtx) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:467]: I don't really like Main as a name for an actor function, it's a bit too GameState-y.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 5]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+    s32 pad;
+
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        EnKusa_SetupLiftedUp(this);
+        SoundSource_PlaySfxAtFixedWorldPos(globalCtx, &this->actor.world.pos, 0x14, NA_SE_PL_PULL_UP_PLANT);
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:472]: I would guess that's a timer?

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 6]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+        this->actor.shape.shadowDraw = ActorShadow_DrawCircle;
+
+    } else if (this->collider.base.acFlags & AC_HIT) {
+        this->collider.base.acFlags &= ~AC_HIT;
+        EnKusa_SpawnFragments(this, globalCtx);
+        EnKusa_DropCollectible(this, globalCtx);
+        SoundSource_PlaySfxAtFixedWorldPos(globalCtx, &this->actor.world.pos, 0x14, NA_SE_EV_PLANT_BROKEN);
+
+        if ((this->actor.params >> 4) & 1) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:481]: And yet more.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 7]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+            if (GET_KUSA_TYPE(this) != 3) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:482]: Missed this enum?

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 8]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+    Vec3f contactPos;
+    s32 i;
+    s16 angleOffset;
+
+    wasHit = (this->collider.base.atFlags & AT_HIT) != 0;
+    if (wasHit) {
+        this->collider.base.atFlags &= ~AT_HIT;
+    }
+    this->timer++;
+    if (((this->actor.bgCheckFlags & 0xB)) || (wasHit) || ((this->timer) >= 100)) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:571]:
```suggestion
    if (((this->actor.bgCheckFlags & 0xB)) || wasHit || ((this->timer) >= 100)) {
```

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 9]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+        if ((this->actor.bgCheckFlags & 0x20) == 0) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:572]:
```suggestion
        if (this->actor.bgCheckFlags & 0x20) {
```
Don't need == on flag checks.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 10]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+    this->timer = 0;
+}
+
+/**
+ * @brief When cut, grass leaves behind a stump that will begin the regrow process after 120 frames
+ *
+ */
+void EnKusa_CutWaitRegrow(EnKusa* this, GlobalContext* globalCtx) {
+    this->timer++;
+    if ((this->timer) >= 120) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:648]:
```suggestion
    if (this->timer >= 120) {
```

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 11]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+    this->actionFunc = EnKusa_UprootedWaitRegrow;
+}
+
+/**
+ * @brief Window of 50 frames before the grass surfaces to then regrow
+ *
+ */
+void EnKusa_UprootedWaitRegrow(EnKusa* this, GlobalContext* globalCtx) {
+    this->timer++;
+    if ((this->timer) > 120) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:676]:
```suggestion
    if (this->timer > 120) {
```

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 12]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+        if ((Math_StepToF(&this->actor.world.pos.y, this->actor.home.pos.y, 0.6f)) && (this->timer >= 170)) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:677]:
```suggestion
        if (Math_StepToF(&this->actor.world.pos.y, this->actor.home.pos.y, 0.6f) && (this->timer >= 170)) {
```

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 13]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+    }
+}
+
+void EnKusa_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    EnKusa* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+
+    this->actionFunc(this, globalCtx);
+
+    if (this->isCut != 0) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:709]: Boolean?
```suggestion
    if (this->isCut) {
```

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 14]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+        Gfx_DrawDListOpa(globalCtx, gameplay_field_keep_DL_0078A0);
+    } else if (this->actor.projectedPos.z < 1300.0f) {
+        s32 alpha;
+        OPEN_DISPS(globalCtx->state.gfxCtx);
+        alpha = (1300.0f - this->actor.projectedPos.z) * 2.55f;
+        func_8012C2DC(globalCtx->state.gfxCtx);
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, alpha);
+        gSPDisplayList(POLY_XLU_DISP++, gameplay_field_keep_DL_007938);
+        CLOSE_DISPS(globalCtx->state.gfxCtx);
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:738]: I'd space this out rather more.
```suggestion
        s32 alpha;

        OPEN_DISPS(globalCtx->state.gfxCtx);

        alpha = (1300.0f - this->actor.projectedPos.z) * 2.55f;
        func_8012C2DC(globalCtx->state.gfxCtx);
        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, alpha);
        gSPDisplayList(POLY_XLU_DISP++, gameplay_field_keep_DL_007938);

        CLOSE_DISPS(globalCtx->state.gfxCtx);
```
And document that displaylist.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 15]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+    }
+}
+
+void EnKusa_DrawGrass(Actor* thisx, GlobalContext* globalCtx) {
+    EnKusa* this = THIS;
+    if (this->isCut != 0) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:744]:
```suggestion
    EnKusa* this = THIS;

    if (this->isCut) {
```

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.h [hunk 16]
```diff
@@ -7,12 +7,29 @@ struct EnKusa;
...
+    /* 0x195 */ s8 objIndex;
+    /* 0x196 */ s8 kusaMtxIdx;
+    /* 0x197 */ u8 isCut;
+    /* 0x198 */ u8 unk198;
 } EnKusa; // size = 0x19C
 
 extern const ActorInit En_Kusa_InitVars;
 
+#define GET_KUSA_COLLECTIBLE_ID(x) (((x->actor.params >> 8) & 0x7F))
+#define GET_KUSA_RAND_COLLECTIBLE_ID(x) ((x->actor.params >> 8) & 0xF)
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.h:31]:
```suggestion
#define GET_KUSA_COLLECTIBLE_ID(x) ((((x)->actor.params >> 8) & 0x7F))
#define GET_KUSA_RAND_COLLECTIBLE_ID(x) (((x)->actor.params >> 8) & 0xF)
```
This is safer, although these should not be passed anything weird. However, we usually make the macros to take `thisx` for consistency and to allow use fo them in the main 4 if necessary.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.h [hunk 17]
```diff
@@ -7,12 +7,29 @@ struct EnKusa;
...
-    /* 0x0144 */ EnKusaActionFunc actionFunc;
-    /* 0x0148 */ char unk_148[0x54];
+    /* 0x000 */ Actor actor;
+    /* 0x144 */ EnKusaActionFunc actionFunc;
+    /* 0x148 */ ColliderCylinder collider;
+    /* 0x194 */ u8 timer;
+    /* 0x195 */ s8 objIndex;
+    /* 0x196 */ s8 kusaMtxIdx;
+    /* 0x197 */ u8 isCut;
+    /* 0x198 */ u8 unk198;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.h:25]: Looks like this is testing something to do with water?

**Review by EllipticEllipsis — COMMENTED** [review]
Good job on the decompolation and documentation so far, that matrix function looks nasty. I've got quite a few comments, but most of them are small style suggestions. I do take issue with some of the names: I remarked on `Main`, but I'd also like clarification on grass vs bush: really these are all bushes, right?

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 18]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+/**
+ * @brief Updates the state of the provided matrix
+ *
+ * @param[in]     matrix  Matrix to update the current state
+ *
+ */
+void EnKusa_GetMtxState(MtxF* matrix) {
+    MtxF* mtxState = Matrix_GetCurrentState();
+    f32* tmp = (f32*)&mtxState->mf[0];
+    f32* tmp2 = (f32*)&matrix->mf[0];
```
[the remark this answers is not in the document]
Comment (jpburnett) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:122]: It still gives OK if I remove them, if I remove them it creates a warning, so I kept the casts in.

[the remark this answers is not in the document]
Comment (hensldm) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:122]: What happens if you try 
```suggestion
    f32* tmp = &mtxState->mf[0][0];
    f32* tmp2 = &matrix->mf[0][0];
```

[the remark this answers is not in the document]
Comment (jpburnett) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:122]: That indeed works. Thanks!

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 19]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+}
+
+/**
+ * @brief Creates a small scale version of grass bush
+ *
+ */
+void EnKusa_SetScaleSmall(EnKusa* this) {
+    this->actor.scale.y = 0.16000001f;
+    this->actor.scale.x = 0.120000005f;
+    this->actor.scale.z = 0.120000005f;
```
[the remark this answers is not in the document]
Comment (jpburnett) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:281]: Not from what I have tried. I have tried fractions and shortening the floats, but that hasn't worked. Other suggestions?

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 20]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+        EnKusa_Sway();
+        kusaGameplayFrames = globalCtx->gameplayFrames;
+    }
+    this->kusaMtxIdx = D_809366B0 & 7;
+    D_809366B0 += 1;
+}
+
+void EnKusa_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    EnKusa* this = THIS;
```
[the remark this answers is not in the document]
Comment (jpburnett) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:425]: Yeah :/

**Comment by jpburnett** [comment]
In regards to clarification on grass and bush, I felt as if there was enough distinction between the two. The left being more like a bush, and the right being more like grass. I would be interested in other thoughts though: 
image

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 21]
```diff
@@ -55,72 +111,655 @@ static InitChainEntry D_80936754[] = {
...
+                }
+            }
+        }
+    }
+}
+
+void EnKusa_SetupLiftedUp(EnKusa* this) {
+    this->actionFunc = EnKusa_LiftedUp;
+    this->actor.room = -1;
+    this->actor.flags |= 0x10;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:520]:
```suggestion
    this->actor.flags |= ACTOR_FLAG_10;
```

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 22]
```diff
@@ -55,72 +111,655 @@ static InitChainEntry D_80936754[] = {
...
+    if (isFullyGrown) {
+        Actor_SetScale(&this->actor, 0.4f);
+        EnKusa_SetupInteract(this);
+        this->collider.base.ocFlags1 &= ~OC1_TYPE_PLAYER;
+    }
+}
+
+void EnKusa_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    EnKusa* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:708]:
```suggestion
    GlobalContext* globalCtx = globalCtx2;
    EnKusa* this = THIS;
```
Would be preferable if it still matches.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c (hunk 11 above)
Comment (hensldm) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:679]: Doesn't look to have been addressed. Forgot to commit/push?

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c (hunk 10 above)
Comment (hensldm) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:651]: Doesn't look to have been addressed. Forgot to commit/push?

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c (hunk 8 above)
Comment (hensldm) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:574]: Doesn't look to have been addressed. Forgot to commit/push?

**Review by hensldm — COMMENTED** [review]
Some thoughts.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c (hunk 8 above)
Comment (jpburnett) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:574]: This should have been pushed up, is it not showing up?

Comment (hensldm) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:574]: Yeah, I still see the parens in the PR.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 23]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kusa/EnKusa_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kusa/func_80936414.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kusa/func_809365CC.s")
+/**
+ * @brief Updates the state of the provided matrix
+ *
+ * @param[in]     matrix  Matrix to update the current state
+ *
```
[the remark this answers is not in the document]
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c]: Was this actually done everywhere?

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 24]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+    }
+}
+
+/**
+ * @brief Spawns bugs
+ *
+ */
+void EnKusa_SpawnBugs(EnKusa* this, GlobalContext* globalCtx) {
+    u32 numBugs = 0;
+    for (numBugs = 0; numBugs < 3; numBugs++) {
```
[the remark this answers is not in the document]
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c]: Does it set it twice?

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 25]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+        this->actor.shape.shadowScale = 0.9f;
+        this->actor.shape.shadowAlpha = 0x46;
+    }
+
+    if (this->actor.shape.rot.y == 0) {
+        this->actor.shape.rot.y = ((u32)Rand_Next() >> 0x10);
+        this->actor.home.rot.y = this->actor.shape.rot.y;
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+    }
+    if (EnKusa_SnapToFloor(this, globalCtx, 0.0f) == 0) {
```
[the remark this answers is not in the document]
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:397]: Did this get missed?

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 26]
```diff
@@ -55,72 +111,656 @@ static InitChainEntry D_80936754[] = {
...
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnKusa_SetupWaitObject(EnKusa* this) {
+    this->actionFunc = EnKusa_WaitObject;
+}
+
+/**
+ * @brief Waits for object to be loaded then draws the appropriate model
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:438]: This function doesn't actually draw anything, though. I think better to just remove this one too.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 27]
```diff
@@ -55,72 +111,656 @@ static InitChainEntry D_80936754[] = {
...
+        SoundSource_PlaySfxAtFixedWorldPos(globalCtx, &this->actor.world.pos, 20, NA_SE_PL_PULL_UP_PLANT);
+        this->actor.shape.shadowDraw = ActorShadow_DrawCircle;
+
+    } else if (this->collider.base.acFlags & AC_HIT) {
+        this->collider.base.acFlags &= ~AC_HIT;
+        EnKusa_SpawnFragments(this, globalCtx);
+        EnKusa_DropCollectible(this, globalCtx);
+        SoundSource_PlaySfxAtFixedWorldPos(globalCtx, &this->actor.world.pos, 20, NA_SE_EV_PLANT_BROKEN);
+
+        if ((KUSA_GET_PARAMS_04(&this->actor))) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:485]: Well, surely you can name this one more descriptively? `KUSA_GET_SHOULD_SPAWN_BUGS`? It's used once so it's okay for it to be a bit longer.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 28]
```diff
@@ -55,72 +111,656 @@ static InitChainEntry D_80936754[] = {
...
+    this->actionFunc = EnKusa_Fall;
+    rotSpeedXtarget = -0xBB8;
+    rotSpeedYtarget = (Rand_ZeroOne() - 0.5f) * 1600.0f;
+    rotSpeedX = 0;
+    rotSpeedY = 0;
+    this->timer = 0;
+}
+
+/**
+ * @brief Handles logic for when grass/bush falls
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:560]: Again, don't think this really adds anything, unless you want to go into more detail.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c (hunk 26 above)
Comment (jpburnett) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:438]: While it doesn't draw, it calls `this->actor.draw` so would a better name be `EnKusa_SetupDraw`?

**Review by hensldm — APPROVED** [review]

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 29]
```diff
@@ -55,72 +111,624 @@ static InitChainEntry D_80936754[] = {
...
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kusa/EnKusa_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kusa/func_80936414.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kusa/func_809365CC.s")
+/**
+ * @brief Applies a "swaying" motion of the provided matrix
+ *
+ * @param[in]     matrix  Matrix to update the current state
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:117]: Looks like you missed some.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 30]
```diff
@@ -55,72 +111,624 @@ static InitChainEntry D_80936754[] = {
...
+    if (this->actor.velocity.y < this->actor.terminalVelocity) {
+        this->actor.velocity.y = this->actor.terminalVelocity;
+    }
+}
+
+/**
+ * @brief Scales a vector down by provided scale factor
+ *
+ * @param vec   vector to be scaled
+ * @param scaleFactor   scale factor to be applied to vector
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:275]: Here too.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c (hunk 25 above)
[the remark this answers is not in the document]
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:389]: Did this get missed?

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 31]
```diff
@@ -55,72 +111,624 @@ static InitChainEntry D_80936754[] = {
...
+            rotSpeedX >>= 1;
+            rotSpeedXtarget >>= 1;
+            rotSpeedY >>= 1;
+            rotSpeedYtarget >>= 1;
+            this->actor.bgCheckFlags &= ~0x40;
+            SoundSource_PlaySfxAtFixedWorldPos(globalCtx, &this->actor.world.pos, 40, NA_SE_EV_DIVE_INTO_WATER_L);
+        }
+        EnKusa_UpdateVelY(this);
+        Math_StepToS(&rotSpeedX, rotSpeedXtarget, 0x1F4);
+        Math_StepToS(&rotSpeedY, rotSpeedYtarget, 0xAA);
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:601]: I think speeds in dec. Also, do they have to be shifts above?

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 32]
```diff
@@ -55,72 +111,624 @@ static InitChainEntry D_80936754[] = {
...
+    this->actionFunc = EnKusa_Regrow;
+    EnKusa_SetScaleSmall(this);
+    this->isCut = false;
+    this->actor.shape.rot = this->actor.home.rot;
+}
+
+void EnKusa_Regrow(EnKusa* this, GlobalContext* globalCtx) {
+    s32 isFullyGrown;
+
+    isFullyGrown = Math_StepToF(&this->actor.scale.y, 0.4f, 0.014f) & 1;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:665]: Can this start with an `= 1` line and then keep `&`ing?

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.h [hunk 33]
```diff
@@ -7,12 +7,33 @@ struct EnKusa;
 
 typedef void (*EnKusaActionFunc)(struct EnKusa*, GlobalContext*);
 
+typedef enum {
+    /* 0 */ ENKUSA_TYPE_BUSH,
+    /* 1 */ ENKUSA_TYPE_REGROW_GRASS,
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.h:12]:
```suggestion
    /* 1 */ ENKUSA_TYPE_REGROWING_GRASS,
```
I know it's longer, but the ungrammatical version's bothering me more.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c (hunk 23 above)
[the remark this answers is not in the document]
Comment (jpburnett) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c]: should be fixed now (I think)

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c (hunk 31 above)
Comment (jpburnett) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:601]: Yes the shifts are needed

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 34]
```diff
@@ -270,7 +267,6 @@ void EnKusa_UpdateVelY(EnKusa* this) {
 
 /**
  * @brief Scales a vector down by provided scale factor
- *
  * @param vec   vector to be scaled
  * @param scaleFactor   scale factor to be applied to vector
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:271]: I meant between the argument name and the description, it's fine to have a blank line before arguments.

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 35]
```diff
@@ -55,72 +111,623 @@ static InitChainEntry D_80936754[] = {
...
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kusa/func_80936414.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kusa/func_809365CC.s")
+/**
+ * @brief Applies a "swaying" motion of the provided matrix
+ *
+ * @param[in]     matrix  Matrix to update the current state
+ */
+void EnAm_ApplySway(MtxF* matrix) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:119]: Typo?

**Review by EllipticEllipsis — APPROVED** [review]

File: src/overlays/actors/ovl_En_Kusa/z_en_kusa.c [hunk 36]
```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
...
+
+        ptr[4] = tempf3;
+        ptr[5] = sp7C[0];
+        ptr[6] = tempf3;
+        ptr[7] = 0.0f;
+
+        ptr[8] = tempf4;
+        ptr[9] = tempf5;
+        ptr[10] = sp7C[3] * 0.2f;
+        ptr[11] = 0.0f;
```
[the remark this answers is not in the document]
Comment (Kenix3) [reply src/overlays/actors/ovl_En_Kusa/z_en_kusa.c:203]: I think the angles are meant to simulate wind.

**Review by Kenix3 — APPROVED** [review]
Good documentation on this actor! Check out my other comment for some thoughts on the unnamed data.

**Pull request merged**